packages/next/build/webpack-config.ts

```diff
@@ -839,6 +841,7 @@ export default async function getBaseWebpackConfig(
         }),
       !isServer &&
         useTypeScript &&
+        !typescriptTranspileOnly &&
```

Comment:
This should still fail during `next build` IMO -- this option should only be disabling development mode checks.

Reply:
From https://github.com/zeit/next.js/issues/8331 by @Timer:

> This flag would only disable the forked type checking process (dev and build). tsconfig.json generation would still be mandatory.

😃

I would personally prefer to disable checks both in `dev` and `build`. As a team that relies on running `yarn lint:tsc`, this is what we do now [via a workaround](https://github.com/zeit/next.js/issues/7687#issuecomment-506440999) anyway. Enabling builds can be useful: imagine a repo with CI/CD configured to deploy preview apps out of PRs. The developers are better off if they can see the result live despite a couple of minor TS errors. The PR will be still unmergeable until `yarn lint` (another CI job) succeeds.

Reply:
Disabling type checks during build would strictly be an optimization, right?

If that's the case, I strongly believe disabling build-time type-checks should be a separate option added after this PR. Also, appropriately named along the lines `dangerouslyEmitOnError`.

Reply:
Not sure I understand the difference between the dev mode and the build when it comes to skipping TypeScript checks. To me, `{ typescript: { transpileOnly: true }}` is the same as using [`ts-node --transpile-only`](https://github.com/TypeStrong/ts-node#cli-and-programmatic-options). If I want to opt-in, I probably have `tsc --noEmit` somewhere amongst my linting scripts and CI anyway, so I want Next.js to ignore TypeChecking this in both cases.

Removing TS checks during build is not an optimisation, it's an enabler to get a working deployed app even when there are type-related imperfections. This is useful for reviewing incomplete PRs.

Looking at prior discussions (#7687, #7809, #7848, #8331), I can't recall requests to distinguish dev and build.

Reply:
Actually, there is still a way to skip checks only during development 💡

```js
const { PHASE_DEVELOPMENT_SERVER } = require('next/constants')

module.exports = (phase, { defaultConfig }) => {
  if (phase === PHASE_DEVELOPMENT_SERVER) {
    return {
      ...defaultConfig,
      typescript: {
        transpileOnly: true
      }
    }
  }
  return defaultConfig
}
```

@Timer WDYT?

Reply:
I'm very confident after landing this option users will ask for a way to split this between development and production, and the above work-around seems needlessly complex.

I'm still in favor of splitting the options.

Reply:
Hmmm is there any evidence suggesting that this may be the case? The demand for the option has been around for about four months, but I cannot remember anyone mentioning the split between dev and build.

If we add two options, what would be the names?

@timneutkens what are your thoughts here?

Reply:
Most of the demand for this option existed before we stopped reporting errors for inactive pages.

Additionally, the verbiage used by users was mostly along the development lines:

- OP in #7848: only complaining about the development experience. Also complains about invalid pages being reported, which was fixed.
- > [It kind of breaks the development flow for me.](https://github.com/zeit/next.js/issues/7848#issuecomment-509967269)
- > [doesn't get in the way of development](https://github.com/zeit/next.js/issues/7687#issuecomment-512080577)
- [An option to disable ForkTsCheckerWebpackPlugin on dev would be awesome](https://github.com/zeit/next.js/issues/7809#issuecomment-519075208)
- [not practical to have Next run type checking during each file edit.](https://github.com/zeit/next.js/issues/8331#issuecomment-521502187)

This is not to say no one has requested a build-time opt-out as well, notably you and https://github.com/zeit/next.js/issues/7687#issuecomment-517712299.

I feel it's important to give users the correct granularity, especially when configuration is easily shared and can be copy/pasted.
Users will likely opt for the quick-to-copy example instead of the more involved edit of their configuration to support phases.

Reply:
As for names, maybe:
```js
{
  typescript: {
    transpileOnly: false, // disables development type checking
    buildOnError: false // disabled production type checking (`next build`)
  }
}
```

Reply:
How about 

```js
{
  typescript: {
    ignoreDevErrors: false,
    ignoreBuildErrors: false
  }
}
```

`transpileOnly` reminds `ts-node --transpile-only`, which may be interpreted as 'this applice to any phase'.

I'm still in favour of a single option, but would not mind refactoring to two options if we agree. Waiting for @timneutkens's thoughts following his [approval of my original proposal](https://github.com/zeit/next.js/issues/8331#issuecomment-543791654).

Reply:
@kachkaev that sound good 👍

Reply:
@timneutkens just to confirm, are you referring to `ignoreDevErrors` / `ignoreBuildErrors`? If both you and @Timer agree on this, I'll be happy to make corresponding changes in the PR 🙂

Reply:
Yeah the naming sounds fine 👍

Reply:
Thanks for insisting on the change @Timer! At first, I was not very happy about your suggestion and and planned to make the changes just to get the PR through. After spending more time with the new flags, I now genuinely feel that this is a much better way to go than just with `transpileOnly` 🎉  Great call 🙌